rust/agents/relayer/src/relayer.rs

```diff
@@ -391,18 +393,21 @@ impl Relayer {
 
         let span = info_span!("MessageProcessor", origin=%message_processor.domain());
         let processor = Processor::new(Box::new(message_processor));
+
+        let origin = origin.clone();
         tokio::spawn(async move {
-            let res = tokio::try_join!(processor.spawn())?;
-            info!(?res, "try_join finished for message processor");
-            Ok(())
+            // Propagate task panics
+            processor.spawn().await.unwrap_or_else(|err| {
```

Comment:
the pattern of spawning a task that spawns a task but awaits on it feels a little awkward, would we get the same behavior if we just returned `processor.spawn()` and instrumented it so we know which task it is that had issues?

don't feel strongly

Reply:
`processor .spawn()` returns a Result though, so we need to do this to restrict other tasks from propagating errors

Reply:
Hmm I don't get it, I think after your changes spawn returns a JoinHandle and not a Result, then awaiting it converts that JoinHandle to a Result that Errs only if there's a panic in the spawned task. And then `run_message_processor` then returns a JoinHandle that will also Err if there's a panic on line 401

So I think the logic rn is similar to the following in Typescript:

try {
  await processor.spawn()
} catch (err) {
  console.log('there was an error')
  throw err
}

which is kinda superfluous except for the console.log adding more info?

Reply:
ah finally got what you meant. Yep so we lose the origin log but it's much cleaner now 👍